phpBB/includes/functions_privmsgs.php

```diff
@@ -888,6 +888,7 @@ function update_unread_status($unread, $msg_id, $user_id, $folder_id)
 	$sql = 'UPDATE ' . PRIVMSGS_TO_TABLE . "
 		SET pm_unread = 0
 		WHERE msg_id = $msg_id
+			AND pm_unread = 1
```

Comment:
Please move this below the `AND folder_id`. It should be marginally faster if it's at bottom cause it's faster at selecting the proper primary key in this query.